LayoutTests/fullscreen/full-screen-keyboard-lock-option.html

```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>This tests that full screen mode with keyboard lock option allows ESC key event.</title>
+</head>
+<body>
+    <div width: 100%; height: 100%;" id="target">
+        <p>Test passes if ESC key received by the test case after enter to fullscreen.</p>
+    </div>
+    <script src="full-screen-test.js"></script>
+    <script>
+
+        window.internals.settings.setFullScreenKeyboardLock(false);
+
+        var callback;
```

Comment:
I think the tests could be simplified a lot... they seem to be using old JS idioms... I haven't tried this, but could you rewrite this test and the other more as:
```JS
window.internals.settings.setFullScreenKeyboardLock(true);
var event;
runWithKeyDown(async () => {
    try {
        const target = document.querySelector("#target");
        await target.requestFullscreen({ keyboardLock: "browser" });
        testExpected("document.fullscreenElement", true);
        event = await new Promise((resolve) => {
            waitForEvent(document, "keypress", resolve, /*endit=*/ false, /*once=*/ true);
            eventSender.keyDown("escape");
        });
        testExpected("event.keyCode", 27); // 'Escape'
    } catch (err) {
        logResult(
            false,
            `An error occurred while trying to enter full-screen mode: ${err.message}`
        );
    } finally {
        endTest();
    }
});
```
I think there is also a way of not having to check the global `var event` either...